dotenv/main.py

```diff
@@ -181,21 +181,42 @@ def unset_key(dotenv_path, key_to_unset, quote_mode="always"):
 
 
 def resolve_nested_variables(values):
-    def _replacement(name):
+    def _replacement(name, default=None):
```

Comment:
How about some unit tests?

Reply:
If you like the general idea, i can go on and write the tests and expand the documentation with examples and caveats.

Reply:
If you like the general idea, i can go on and write tests and expand the documentation with the new functionality and some caveats.

Reply:
tests would help to clarify what the general idea is.

Note that I'm not a contributor to this library, so my opinion doesn't carry any weight.